src/crystal/system/unix/group.cr

```diff
@@ -0,0 +1,58 @@
+require "c/grp"
+
+module Crystal::System::Group
+  private GETGR_R_SIZE_MAX = 1024 * 16
+
+  private def extract_members(gr_mem)
+    members = Array(String).new
+
+    n = 0
+    while gr_mem[n]
+      members << String.new(gr_mem[n])
+      n += 1
+    end
+
+    members
+  end
+
+  private def from_struct(grp)
+    new(String.new(grp.gr_name), grp.gr_gid.to_s, extract_members(grp.gr_mem))
+  end
+
+  private def from_name?(groupname : String)
+    groupname.check_no_null_byte
+
+    grp = uninitialized LibC::Group
+    grp_pointer = pointerof(grp)
+    initial_buf = uninitialized UInt8[1024]
+    buf = initial_buf.to_slice
+
+    ret = LibC.getgrnam_r(groupname, grp_pointer, buf, buf.size, pointerof(grp_pointer))
```

Comment:
`out grp` could be used here and in other places https://crystal-lang.org/reference/syntax_and_semantics/c_bindings/out.html

Reply:
Yep, this is what I ran into.